include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -557,6 +557,18 @@ typedef struct OrtOpenVINOProviderOptions {
   unsigned char enable_dynamic_shapes;  ///< 0 = disabled, nonzero = enabled
 } OrtOpenVINOProviderOptions;
 
+/** \brief OneDNN Provider Options
+ *
+ * \see OrtApi::SessionOptionsAppendExecutionProvider_Dnnl
+ */
+typedef struct OrtDnnlProviderOptions {
+#ifdef __cplusplus
+  OrtDnnlProviderOptions() : use_arena{true}, threadpool_args{nullptr} {}
+#endif
+  int use_arena;          // If arena is used, use_arena 0 = not used, nonzero = used
```

Comment:
Please avoid exposing data structures. Use get/set methods instead.  Otherwise whenever you need to make changes to this data type, it will break ORT's binary compatibility.

Reply:
ORT maintains binary compatibility for C API (i.e. applications linked against libonnxruntime should expect to continue working with newer versions)
Consider using an opaque struct instead, so fields can be updated without impact to binary compatibility.
see comments at
https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L3232
and
https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L2743
the caller does not directly access the struct.
instead, they call an interface to create the struct, update it, and release it when they're done

Reply:
@snnn @jywu-msft could you please check current changes and verify that current implementation is what you expected